mod/01_core/format/html_format.rb

```diff
@@ -216,8 +265,8 @@ def unsupported_view view
     end
 
     def final_link href, opts={}
-      text = opts[:text] || href
-      %{<a class="#{opts[:class]}" href="#{href}">#{text}</a>}
+      text = (opts.delete(:text) || href).dup
+      content_tag :a, text, opts.merge(:href=>href)
```

Comment:
Note that the options from build_link are passed here, and now they are all passed to content_tag as options.  They tests would be failing otherwise and they aren't.

This is also where the & get's html encoded.  We could try to do something here to prevent that, but I think that would actually be wrong.  It isn't clear to me why link_to doesn't similarly process the & characters, but the question still remains if the new behavior is broken in any functional way.

If you make the case that it is broken, I'd be happy to try to fix it, but I'm still not convinced it is.  I think it was an oversight with how it did work, and I think it is a bad thing to have more than one piece of code generating final links.